packages/block-editor/src/store/actions.js

```diff
@@ -1445,13 +1445,19 @@ export const __unstableSetEditorMode =
 				const sectionClientIds =
 					select.getBlockOrder( sectionRootClientId );
 				if ( sectionClientIds ) {
-					const parents = select.getBlockParents(
-						firstSelectedClientId
-					);
-					const firstSectionClientId = parents.find( ( parent ) =>
-						sectionClientIds.includes( parent )
-					);
-					dispatch.selectBlock( firstSectionClientId );
+					if ( firstSelectedClientId ) {
+						const parents = select.getBlockParents(
+							firstSelectedClientId
+						);
+						const firstSectionClientId = parents.find( ( parent ) =>
+							sectionClientIds.includes( parent )
+						);
+						dispatch.selectBlock( firstSectionClientId );
+					} else {
+						const lastChildSectionClientId =
```

Comment:
```suggestion
						const lastSectionClientId =
```

Reply:
Why remove the word child? My understanding is that this block is the last one in the main section...

Reply:
The `main` tag group, or post content, or root element are the section root containers, and their immediate children are **sections**. We do not have the concept of main section and child sections. Main is only the tag name we conventionally use.

Reply:
Ah ok, thanks :)